### PR TITLE
Refactor the read-column-thunk for Types/TIMESTAMP_WITH_TIMEZONE method

### PR DESCRIPTION
**Pull request by lpoulain**
This PR does not change the behavior of the handler but refactors it.

### CODE REVIEW

**Comment by hovaesco** [comment]
Tests are failing

File: drivers/starburst/src/metabase/driver/implementation/execute.clj [hunk 1]
```diff
@@ -121,11 +121,12 @@
 
 (defmethod sql-jdbc.execute/read-column-thunk [:starburst Types/TIMESTAMP_WITH_TIMEZONE]
   [_ ^ResultSet rset _ ^long i]
-  ;; Converts TIMESTAMP_WITH_TIMEZONE to instant, then to OffsetDateTime with default time zone.
+  ;; Converts TIMESTAMP_WITH_TIMEZONE to java.time.ZonedDateTime with default time zone.
```
Comment (andrewdibiasio6) [inline drivers/starburst/src/metabase/driver/implementation/execute.clj:124]: is this comment true? Are are converting it to a `OffsetDateTime` in the end? Maybe we can update comment with updated details as to what's happening and why.

File: drivers/starburst/src/metabase/driver/implementation/execute.clj [hunk 2]
```diff
@@ -121,11 +121,12 @@
...
   (fn []
-    (when-let [t (.getObject rset i java.sql.Timestamp)] 
-      (let [instant (.toInstant t)] 
-        (.atOffset instant (t/zone-offset 0))))))
+    (let [zonedDateTime (.getObject rset i java.time.ZonedDateTime)
+          utcTimeZone (java.time.ZoneId/of "UTC")]
+      (.toOffsetDateTime (.withZoneSameInstant zonedDateTime utcTimeZone))
+    )))
```
Comment (andrewdibiasio6) [inline drivers/starburst/src/metabase/driver/implementation/execute.clj:129]: so is the idea here is we are creating a zonedDateTime, which means we shouldn't lose any info about the time zone, and then we are setting the time zone to UTC, then converting to offsetDateTime? 

Is this related to fixing the [bug](https://github.com/starburstdata/metabase-driver/issues/58) here? 

Did this end up fixing the issue? Can you add some testing details to the PR overview that shows it is fixed?

Comment (lpoulain) [reply drivers/starburst/src/metabase/driver/implementation/execute.clj:129]: The bug is actually not a bug. It's PostgreSQL which exhibits some weird behavior. But we confirmed with Metabase that the Starburst driver has the right behavior.

This PR is a refactor which should not affect the behavior.

File: drivers/starburst/src/metabase/driver/implementation/execute.clj (hunk 1 above)
Comment (lpoulain) [reply drivers/starburst/src/metabase/driver/implementation/execute.clj]: Good catch. Fixed.

**Comment by andrewdibiasio6** [comment]
In general, I am not sure what the goal of this PR is. Is it fixing or related to  https://github.com/starburstdata/metabase-driver/pull/60 in any way or is it just a general clean up?

**Comment by lpoulain** [comment]
@andrewdibiasio6 commits squashed. As far as the chances it's a general clean up. I'll let @aalbu comment further.

**Review by andrewdibiasio6 — APPROVED** [review]
Looks cleaner to me!

**Pull request merged**